### PR TITLE
Adding description of version metadata setting

### PR DESCRIPTION
**Pull request by john-science**
## Description

The metadata settings section was not described in the docs, so this PR gives a quick description of the only field in that section so far.

---

## Checklist

- [X] This PR has only one purpose or idea.
- [X] Tests have been added/updated to verify that the new/changed code works.

- [X] The code style follows [good practices](https://terrapower.github.io/armi/developer/standards_and_practices.html).
- [X] The commit message(s) follow [good practices](https://terrapower.github.io/armi/developer/tooling.html).

- [X] The [release notes](https://terrapower.github.io/armi/release/index.html) (location `doc/release/0.X.rst`) are up-to-date with any bug fixes or new features.
- [X] The documentation is still up-to-date in the `doc` folder.
- [X] The dependencies are still up-to-date in `setup.py`.

### CODE REVIEW

File: doc/tutorials/walkthrough_inputs.rst [hunk 1]
```diff
@@ -353,13 +353,19 @@ Now we need to specify some **settings** that define fundamental reactor
...
 .. literalinclude:: ../../armi/tests/tutorials/anl-afci-177.yaml
     :language: yaml
     :start-after: begin-metadata
     :end-before: begin-settings
 
+Here, the version is "uncontrolled" because this is just a test file we want
+to always work with the latest ARMI. But if you are doing a really important
+analysis, you probably want to specify which version of ARMI you used, like:
+"version: 0.2.5", or similar. Obviously, metadata is for documentation
+purposes, so it's optional. And "uncontrolled" is the preferred default.
```
Comment (keckler) [inline doc/tutorials/walkthrough_inputs.rst:367]: I think it is actually _not_ optional.

Comment (jakehader) [reply doc/tutorials/walkthrough_inputs.rst:367]: I believe that ARMI will not be able to load a settings file if this is not in it.

**Review by jakehader — CHANGES_REQUESTED** [review]
Small comment from @keckler

File: doc/tutorials/walkthrough_inputs.rst (hunk 1 above)
Comment (john-science) [reply doc/tutorials/walkthrough_inputs.rst:367]: Yeah, I was being a little loosey-goosey with my language, that's a good call.

What I was _thinking_ was specifying a real version was "optional"; as opposed to just specifying it as "uncontrolled".

I'll fix it.

Comment (john-science) [reply doc/tutorials/walkthrough_inputs.rst]: @keckler @jakehader Is that better?

**Review by keckler — APPROVED** [review]

**Review by jakehader — APPROVED** [review]

**Pull request merged**